lib/Network/OAuth2/Server/API.hs

```diff
@@ -309,22 +309,22 @@ verifyEndpoint
     -> Maybe AuthHeader
     -> Token
     -> m (Headers '[Header "Cache-Control" NoCache] AccessResponse)
-verifyEndpoint ServerState{..} Nothing _token = throwError
-    err401 { errHeaders = toHeaders challenge
-           , errBody = "You must login to validate a token."
-           }
+verifyEndpoint ServerState{..} Nothing _token =
+    throwError login
   where
-    challenge = BasicAuth $ Realm (optVerifyRealm serverOpts)
+    login = err401 { errHeaders = toHeaders $ BasicAuth (Realm $ optVerifyRealm serverOpts)
+                   , errBody = "Login to validate a token."
+                   }
 verifyEndpoint ServerState{..} (Just auth) token' = do
     -- 1. Check client authentication.
     client_id' <- liftIO . try $ storeCheckClientAuth serverPGConnPool auth
     client_id <- case client_id' of
         Left e -> do
             logE $ "Error verifying token: " <> show (e :: OAuth2Error)
-            throwError err500 { errBody = "Error checking client credentials." }
+            throwError login -- err500 { errBody = "Error checking client credentials." }
```

Comment:
These should be fixed once `storeCheckClientAuth` gets back to returning Nothing when there's no matching client instead of throwing an exception.